src/planet-a/SubdivTriangleMesh.cpp

```diff
@@ -0,0 +1,170 @@
+                chunkMax, subdivLevels, sharedMax, fanMax);
+}
+
+ChunkId ChunkedTriangleMeshInfo::chunk_create(
+        SubdivTriangleSkeleton& rSkel,
+        SkTriId skTri,
+        ArrayView_t<SkVrtxId> const edgeRte,
+        ArrayView_t<SkVrtxId> const edgeBtm,
+        ArrayView_t<SkVrtxId> const edgeLft)
+{
+    if (   edgeRte.size() != m_chunkWidth - 1
+        || edgeBtm.size() != m_chunkWidth - 1
+        || edgeLft.size() != m_chunkWidth - 1)
+    {
+        throw std::runtime_error("Incorrect edge vertex count");
+    }
+
+    // Create a new chunk ID
+    ChunkId const chunkId = m_chunkIds.create();
+    Chunk &rChunk = m_chunkData[size_t(chunkId)];
+
+    rChunk.m_skeletonTri = rSkel.tri_store(skTri);
+
+    SkeletonTriangle const &tri = rSkel.tri_at(skTri);
+
+    ArrayView_t<SharedVrtxStorage_t> const sharedSpace = chunk_shared_mutable(chunkId);
+
+    std::array const edges = {edgeRte, edgeBtm, edgeLft};
+
+    for (int i = 0; i < 3; i ++)
+    {
+        size_t const cornerOffset = m_chunkWidth * i;
+        size_t const edgeOffset = m_chunkWidth * i + 1;
+
+        ArrayView_t<SkVrtxId> const edge = edges[i];
+        {
+            SharedVrtxId const sharedId = shared_get_or_create(
+                    tri.m_vertices[i], rSkel);
+            sharedSpace[cornerOffset] = shared_store(sharedId);
+        }
+        for (unsigned int j = 0; j < m_chunkWidth - 1; j ++)
+        {
+            SharedVrtxId const sharedId = shared_get_or_create(edge[j], rSkel);
+            sharedSpace[edgeOffset + j] = shared_store(sharedId);
+        }
+    }
+
+    return chunkId;
+}
+
+//-----------------------------------------------------------------------------
+
+ChunkVrtxSubdivLUT::ChunkVrtxSubdivLUT(uint8_t subdivLevel)
+{
+    m_edgeVrtxCount = 1u << subdivLevel;
+    m_fillVrtxCount = (m_edgeVrtxCount-2) * (m_edgeVrtxCount-1) / 2;
+
+    m_data.reserve(m_fillVrtxCount);
+
+    // Calculate LUT, this fills m_data
+    fill_tri_recurse({0,               0              },
+                     {0,               m_edgeVrtxCount},
+                     {m_edgeVrtxCount, m_edgeVrtxCount},
+                     subdivLevel);
+
+    // Future optimization:
+    // m_data can be sorted in a way that slightly improves cache locality
+    // by accessing to fill vertices in a more sequential order.
+}
+
+void ChunkVrtxSubdivLUT::subdiv_line_recurse(
+        Vector2us a, Vector2us b, uint8_t level)
+{
+    Vector2us const mid = (a + b) / 2;
+
+    ChunkLocalSharedId out = ChunkLocalSharedId(xy_to_triangular(mid.x() - 1, mid.y() - 2));
+
+    m_data.emplace_back(ToSubdiv{id_at(a), id_at(b), out});
+
+    if (level > 1)
+    {
+        subdiv_line_recurse(a, mid, level - 1);
+        subdiv_line_recurse(mid, b, level - 1);
+    }
+
+}
+
+void ChunkVrtxSubdivLUT::fill_tri_recurse(
+        Vector2us top, Vector2us lft, Vector2us rte, uint8_t level)
+{
+    // calculate midpoints
+    std::array<Vector2us, 3> mid{{
```

Comment:
Seems odd that elsewhere you're able to do std::array const mid = {{}}, but here you don.t

Reply:
This can be made const. Is there enough of a difference that I should prefer one over the other?